src/Utilities/Compiler/Extensions/SourceTextExtensions.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Analyzer.Utilities.Extensions
+{
+    /// <summary>
+    /// Class that contains extensions to <see cref="SourceText"/>.
+    /// </summary>
+    internal static partial class SourceTextExtensions
+    {
+        /// <summary>
+        /// Reads the <paramref name="text"/> contents into a stream and returns the result of calling the
+        /// <paramref name="parser"/> function on that stream.
+        /// </summary>
+        /// <typeparam name="T">Type to deserialize from the <paramref name="text"/>.</typeparam>
+        /// <param name="text">Abstraction for an additional file's contents.</param>
+        /// <param name="parser">Function that will parse <paramref name="text"/> into <typeparamref name="T"/>.</param>
+        /// <returns>Output from <paramref name="parser"/>.</returns>
+        public static T Parse<T>(this SourceText text, Func<StreamReader, T> parser)
```

Comment:
WOuldn't it be better to have this only returning the `StreamReader`?

Reply:
Why would that be better?

Reply:
Nevermind my comment, I don't know why I missed the usings and thought you will return some content and leave the stream open.